pymodbus/server/requesthandler.py

```diff
@@ -40,23 +40,23 @@ def __init__(self, owner, trace_packet, trace_pdu, trace_connect):
             trace_connect,
         )
 
-    def callback_new_connection(self) -> ModbusProtocol:
+    def callback_new_connection(self):
```

Comment:
This is wrong, the callback returns ModbusProtocol.

Reply:
I tried that, but there is one test that uses what should be an abstract method.  Actually, I think it may be possible to entirely remove this method but that's for another PR.

Reply:
Totally remove this method is interesting, I do not really see how since the content depend on the type of transport.

You could make it abstract and modify that one test.

Reply:
For another day; reverted.